modules/services/dnscrypt-proxy.nix

```diff
@@ -0,0 +1,76 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  
+  cfg = config.services.dnscrypt-proxy;
+
+  format = pkgs.formats.toml { };
+
+  configFile = format.generate "dnscrypt-proxy.toml" cfg.settings;
+
+in
+
+{
+  options.services.dnscrypt-proxy = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to enable the dnscrypt-proxy service.";
+    };
+
+    package = mkOption {
+      type = types.path;
+      default = pkgs.dnscrypt-proxy2;
+      defaultText = "pkgs.dnscrypt-proxy2";
+      description = "This option specifies the dnscrypt-proxy package to use";
+    };
+
+    settings = mkOption {
+      type = format.type;
+      
+      default = {
```

Comment:
Can you set this to an empty attrset then set `example` to match the NixOS module?

https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/dnscrypt-proxy2.nix

Reply:
I have been update this PRs based on your review @Enzime